api/proto/teleport/identitycenter/v1/service.proto

```diff
@@ -0,0 +1,126 @@
+// Copyright 2024 Gravitational, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.identitycenter.v1;
+
+import "teleport/identitycenter/v1/identitycenter.proto";
+import "teleport/provisioning/v1/provisioning.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/identitycenter/v1;identitycenterv1";
+
+// IdentityCenterService provides a remote-access API for the Identity Center
+// integration
+service IdentityCenterService {
+  // DescribePrincipal fetches summary information for a given principal
+  rpc DescribePrincipal(DescribePrincipalRequest) returns (PrincipalSummary);
+
+  // ListPrincipals returns paginated list of principal summaries
+  rpc ListPrincipals(ListPrincipalsRequest) returns (ListPrincipalsResponse);
+
+  // ResetPrincipal resets a principal's external ID and forces them to be re-
+  // provisioned into Identity Center
+  rpc ResetPrincipal(ResetPrincipalRequest) returns (ResetPrincipalResponse);
+}
+
+// PrincipalSummary summarizes the current state of a principal, combining
+// provisioning and account assignment info.
+// buf:lint:ignore PAGINATION_REQUIRED
+message PrincipalSummary {
+  // AssignmentSummary holds summary details about a principal's account
+  // assignment state
+  message AssignmentSummary {
+    ProvisioningState status = 1;
+    string external_id = 2;
+    string error = 3;
+  }
+
+  // ProvisioningSummary holds summary details about a principal's SCIM
+  // provisioning state assignment state
+  message ProvisioningSummary {
+    teleport.provisioning.v1.ProvisioningState status = 1;
+    string external_id = 2;
+    string error = 3;
+  }
+
+  // PrincipalName holds the principal's name in Teleport
+  string principal_name = 1;
+
+  // PrincipalType indicates the principal's type, either user or group
+  PrincipalType principal_type = 2;
+
+  // Assignment holds summary details for the principal's account assignment
+  // state
+  AssignmentSummary assignment = 3;
```

Comment:
Is a single account assignment state sufficient for your use-case? A user can be assigned to multiple accounts/permission sets so just wondering if tracking assignment status per assignment (if we even can do this) would be beneficial or an overkill. No strong opinion though.